src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/DoNotDeclareProtectedMembersInSealedTypes.cs

```diff
@@ -49,7 +49,7 @@ public override void Initialize(AnalysisContext context)
                     return;
                 }
 
-                if (!symbol.IsProtected() ||
```

Comment:
I think it would be better to add other extension methods for `IsProtectedAndInternal` and `IsProtectedOrInternal`. If I were to call `IsProtected`, I would expect it to tell me whether the accessibility of the member is *exactly* protected, not some more or less inclusive variant. Even if other callers of `IsProtected` do wish to also include `protected internal` and `private protected`, I think it would make the code-base clearer to change those callers rather than change `IsProtected` to not do what it says on the tin.

Reply:
@NewellClark I personally won't be very surprised if it returned true for `private protected` etc, but we could rename the method to avoid any confusion. So I still think we should revisit other callers (I wouldn't expect to see lots of callers)

Reply:
I gave a look and here are the usages:

- security analyzer (CA5368 - `SetViewStateUserKey`). Tagging @dotpaul for whether this needs to check `private protected` and `protected internal` or not.
- Used in `INamedTypeSymbolExtensions` (the affected extension is `IsStaticHolderType` which is only called by two analyzers: `AvoidUninstantiatedInternalClasses` (CA1812) and `StaticHolderTypesAnalyzer` (CA1052).

Any other usages I'm missing?

Reply:
@Youssef1313 thanks for checking CA5368. I think just `protected` is fine.

Reply:
Alright, so we need to check both CA1812 and CA1052. If any of them needs to work on all `protected` modifier variants,  then we should move `IsAnyProtectedVariant` to an extension method and call it from the analyzers needing it.

Reply:
https://github.com/dotnet/roslyn-analyzers/blob/e9471f4c49a0837913af03ff22fee146b3f30ced/src/Utilities/Compiler/Extensions/INamedTypeSymbolExtensions.cs#L213-L221
@Youssef1313 @dotpaul I just checked `IsStaticHolderType`, and it appears that it definitely does not want to be checking for `protected internal`, so it would not be able to make use of `IsAnyProtectedVariant`. We may want to add a check for `private protected`, though I'm hesitant to make changes to working code.

Reply:
Local change just to this analyzer seems like a good approach for what the issue is fixing. If required, we can file a follow-up issue for auditing other analyzers.